homeassistant/components/iotty/switch.py

```diff
@@ -0,0 +1,181 @@
+"""Implement a iotty Light Switch Device."""
+
+from __future__ import annotations
+
+import logging
+from typing import Any, cast
+
+from iottycloud.device import Device
+from iottycloud.lightswitch import LightSwitch
+from iottycloud.verbs import LS_DEVICE_TYPE_UID
+
+from homeassistant.components.switch import SwitchDeviceClass, SwitchEntity
+from homeassistant.const import EntityCategory
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import IottyConfigEntry
+from .api import IottyProxy
+from .const import DOMAIN
+from .coordinator import IottyDataUpdateCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: IottyConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Activate the iotty LightSwitch component."""
+    _LOGGER.debug("Setup SWITCH entry id is %s", config_entry.entry_id)
+
+    coordinator: IottyDataUpdateCoordinator = config_entry.runtime_data.coordinator
```

Comment:
mypy should be able to infer the type, although it's probably OK to write it out if you think it improves legibility of the code
```suggestion
    coordinator = config_entry.runtime_data.coordinator
```

Reply:
Done